src/Tizen.NUI.Components/Controls/ItemsView/LinearLayouter.cs

```diff
@@ -0,0 +1,439 @@
+
+                Width = SizeDeligate.Size.Width == 0 ? Width : SizeDeligate.Size.Width;
+                Height = SizeDeligate.Size.Height == 0 ? Height : SizeDeligate.Size.Height;
+            }
+
+            colView.UnrealizeItem(SizeDeligate);
+            if (colView.Header != null) colView.UnrealizeItem(colView.Header);
+            if (colView.Footer != null) colView.UnrealizeItem(colView.Footer);
+            
+            //Console.WriteLine("LSH : Layout Size {0} :{0}", Width, Height);
+
+            // pick the StepCandidate.
+            StepCandidate = isHorizontal ? Width : Height;
+            if (StepCandidate == 0) StepCandidate = 1;
+
+            float Current = 0f;
+            for (int i = 0; i < count; i++)
+            {
+                if (colView.SizingStrategy == ItemSizingStrategy.MeasureAll)
+                {
+                    if (i == 0 && Header != null)
+                        ItemSize.Add(HeaderSize);
+                    else if (i == count -1 && Footer != null)
+                        ItemSize.Add(FooterSize);
+                    else ItemSize.Add(StepCandidate);
+                }
+                ItemPosition.Add(Current);
+                if (i == 0 && Header != null)
+                    Current = HeaderSize;
+                else if (i == count -1 && Footer != null)
+                    Current += FooterSize;
+                else Current += StepCandidate;                
+            }
+
+            // colView.UnrealizeItem(SizeDeligate); keep the SizeDeligate for tracking SizeChange.
+            ScrollContentSize = Current;
+            if (isHorizontal) colView.ContentContainer.SizeWidth = ScrollContentSize;
+            else colView.ContentContainer.SizeHeight = ScrollContentSize;
+
+            base.Initialize(view);
+            //Console.WriteLine("Init Done, StepCnadidate{0}, Scroll{1}", StepCandidate, ScrollContentSize);
+        }
+
+        /// <summary>
+        /// This is called to find out where items are lain out according to current scroll position.
+        /// </summary>
+        /// <param name="scrollPosition">Scroll position which is calculated by ScrollableBase</param>
+        /// <param name="force">boolean force flag to layouting forcely.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override void RequestLayout(float scrollPosition, bool force = false)
+        {
+            // Layouting is only possible after once it intialized.
+            if (!IsInitialized) return;
+            int LastIndex = colView.InternalItemSource.Count;
+
+            if (!force && PrevScrollPosition == Math.Abs(scrollPosition)) return;
+            PrevScrollPosition = Math.Abs(scrollPosition);
+
+            if (ItemSizeChanged >= 0)
+            {
+                for (int i = ItemSizeChanged; i < LastIndex; i++)
+                    UpdatePosition(i);
+                ScrollContentSize = ItemPosition[LastIndex - 1] + GetItemSize(LastIndex);
```

Comment:
It throws `ArgumentOutOfRangeException`.
```
I/DOTNET_LAUNCHER( 7032): System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
I/DOTNET_LAUNCHER( 7032):    at System.Collections.Generic.List`1.get_Item(Int32 index)
I/DOTNET_LAUNCHER( 7032):    at Tizen.NUI.Components.LinearLayouter.GetItemSize(Int32 index) in C:\Users\yj34.lee.CORP\source\repos\TizenFX\src\Tizen.NUI.Components\Controls\ItemsView\LinearLayouter.cs:line 382
I/DOTNET_LAUNCHER( 7032):    at Tizen.NUI.Components.LinearLayouter.RequestLayout(Single scrollPosition, Boolean force) in C:\Users\yj34.lee.CORP\source\repos\TizenFX\src\Tizen.NUI.Components\Controls\ItemsView\LinearLayouter.cs:line 199
I/DOTNET_LAUNCHER( 7032):    at Tizen.NUI.Components.ItemsView.OnRelayout(Vector2 size, RelayoutContainer container) in C:\Users\yj34.lee.CORP\source\repos\TizenFX\src\Tizen.NUI.Components\Controls\ItemsView\ItemsView.cs:line 109
I/DOTNET_LAUNCHER( 7032):    at Tizen.NUI.ViewWrapperImpl.DirectorOnRelayout(IntPtr size, IntPtr container) in C:\Users\yj34.lee.CORP\source\repos\TizenFX\src\Tizen.NUI\src\internal\ViewWrapperImpl.cs:line 370
I/DOTNET_LAUNCHER( 7032):    at Tizen.NUI.Application.MainL
I/DOTNET_LAUNCHER( 7032): oop() in C:\Users\yj34.lee.CORP\source\repos\TizenFX\src\Tizen.NUI\src\internal\Application.cs:line 1272
I/DOTNET_LAUNCHER( 7032):    at Tizen.NUI.NUICoreBackend.Run(String[] args) in C:\Users\yj34.lee.CORP\source\repos\TizenFX\src\Tizen.NUI\src\internal\NUICoreBackend.cs:line 187
I/DOTNET_LAUNCHER( 7032):    at Tizen.Applications.CoreApplication.Run(String[] args)
I/DOTNET_LAUNCHER( 7032):    at Tizen.NUI.NUIApplication.Run(String[] args) in C:\Users\yj34.lee.CORP\source\repos\TizenFX\src\Tizen.NUI\src\public\NUIApplication.cs:line 246
I/DOTNET_LAUNCHER( 7032):    at ThemeResourceDemo.Program.Main(String[] args)
```

```suggestion
                ScrollContentSize = ItemPosition[LastIndex - 1] + GetItemSize(LastIndex -1);
```

Reply:
it has to be + 1 of position but lastIndex should be Count - 1. I fixed lastIndex.